core/domain/story_services.py

```diff
@@ -65,6 +67,39 @@ def _migrate_story_contents_to_latest_schema(versioned_story_contents):
         story_schema_version += 1
 
 
+def get_pretest_questions(story_id, exploration_id):
+    """Checks if an exploration is linked to a story and returns prerequisite
+    skills if so.
+
+    Args:
+        story_id: str. The ID of story linked to the exploration.
+        exploration_id: str. ID of the exploration.
+
+    Returns:
+        list(Question). The list of questions which are the pretests for
+            the exploration, if it is linked to a story, else empty list.
+    """
+    story = get_story_by_id(story_id, strict=False)
+    if story is None:
+        return []
+
+    prerequisite_skill_ids = None
+    for node in story.story_contents.nodes:
+        if node.exploration_id == exploration_id:
+            prerequisite_skill_ids = node.prerequisite_skill_ids
+            break
+
+    if prerequisite_skill_ids is None:
```

Comment:
Check this in the controller. Write in the docstring that the caller should ensure that the story exists and that the exploration is part of the story. OK to raise exceptions here if that's not the case, but probably better not to just paper over it with [] -- you want the buggy code to be discovered and fixed.

Reply:
Either way seems fine to me.

Reply:
Done.